### PR TITLE
PD-635 ApplicationRegistration

### PR DESCRIPTION
**Pull request by nathan-px**

### CODE REVIEW

File: content/portworx-install-with-kubernetes/storage-operations/stateful-applications/application-registration.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+description: Backup, restore, and clone stateful applications
+hidden: false
+weight: 2
+---
+
+With Stork 2.4.3 and newer, you can back up and restore customer-specific CRDs. Using this method, you create a Stork custom resource, called an ApplicationRegistration, which registers your CRD with Stork, allowing you to perform a migration, backup, and restore of your CRDs specified resources. 
+
+## Prerequisites
+
+* Your Portworx installation must contain the `libopenstorage/stork:2.4.3-rc2` image. <!-- am I reading this correctly? -->
```
Comment (nathan-px) [inline content/portworx-install-with-kubernetes/storage-operations/stateful-applications/application-registration.md:14]: @disrani-px I have questions here

File: content/portworx-install-with-kubernetes/storage-operations/stateful-applications/application-registration.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+redis       RedisEnterpriseDatabase    app.redislabs.com        v1                             false
+```
+
+## Register a new CRD with Stork
+
+To register a new CRD with Stork, perform the following steps: 
+
+1. Create an `applicationregistration` spec, specifying the following:
+
+  * **metadata.name:** <!-- What does this mean? and where does it go? under just `metadata`? # list of CR for <myappname> -->
```
Comment (nathan-px) [inline content/portworx-install-with-kubernetes/storage-operations/stateful-applications/application-registration.md:51]: @disrani-px and here

File: content/portworx-install-with-kubernetes/storage-operations/stateful-applications/application-registration.md (hunk 1 above)
Comment (disrani-px) [reply content/portworx-install-with-kubernetes/storage-operations/stateful-applications/application-registration.md:14]: Minimum version required is 2.4.3

File: content/portworx-install-with-kubernetes/storage-operations/stateful-applications/application-registration.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  * **resources.PodsPath:** With the path to any pods that must be deleted prior to disabling the customer resource. <!-- this isn't entirely clear to me as well: # specify pod path in case pods has to be deleted for disabling CR -->
+  * **resources.group:** <!-- need an explanation for each of these -->
```
Comment (nathan-px) [inline content/portworx-install-with-kubernetes/storage-operations/stateful-applications/application-registration.md:53]: @disrani-px I took some guesses, but probably faster if you just add a description here:

resources.group:
resources.version:
resources.kind:
resources.keepStatus:
resources.suspendOptions.path:
resources.suspendOptions.type:

File: content/portworx-install-with-kubernetes/storage-operations/stateful-applications/application-registration.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+couchbase   CouchbaseGroup             couchbase.com            v2                             false
+couchbase   CouchbaseRoleBinding       couchbase.com            v2                             false
+couchbase   CouchbaseBackup            couchbase.com            v2                             false
+couchbase   CouchbaseBackupRestore     couchbase.com            v2                             false
+ibm         IBPCA                      ibp.com                  v1alpha1   spec.replicas,int   false
+ibm         IBPConsole                 ibp.com                  v1alpha1   spec.replicas,int   false
+ibm         IBPOrderer                 ibp.com                  v1alpha1   spec.replicas,int   false
+ibm         IBPPeer                    ibp.com                  v1alpha1   spec.replicas,int   false
+redis       RedisEnterpriseCluster     app.redislabs.com        v1                             false
+redis       RedisEnterpriseDatabase    app.redislabs.com        v1                             false
```
Comment (disrani-px) [inline content/portworx-install-with-kubernetes/storage-operations/stateful-applications/application-registration.md:41]:
```suggestion
$ storkctl get applicationregistrations
NAME        KIND                       CRD-NAME                 VERSION    SUSPEND-OPTIONS                       KEEP-STATUS
cassandra   CassandraDatacenter        cassandra.datastax.com   v1beta1    spec.stopped,bool                     false
couchbase   CouchbaseBucket            couchbase.com            v2                                               false
couchbase   CouchbaseCluster           couchbase.com            v2         spec.paused,bool                      false
couchbase   CouchbaseEphemeralBucket   couchbase.com            v2                                               false
couchbase   CouchbaseMemcachedBucket   couchbase.com            v2                                               false
couchbase   CouchbaseReplication       couchbase.com            v2                                               false
couchbase   CouchbaseUser              couchbase.com            v2                                               false
couchbase   CouchbaseGroup             couchbase.com            v2                                               false
couchbase   CouchbaseRoleBinding       couchbase.com            v2                                               false
couchbase   CouchbaseBackup            couchbase.com            v2                                               false
couchbase   CouchbaseBackupRestore     couchbase.com            v2                                               false
ibm         IBPCA                      ibp.com                  v1alpha1   spec.replicas,int                     false
ibm         IBPConsole                 ibp.com                  v1alpha1   spec.replicas,int                     false
ibm         IBPOrderer                 ibp.com                  v1alpha1   spec.replicas,int                     false
ibm         IBPPeer                    ibp.com                  v1alpha1   spec.replicas,int                     false
redis       RedisEnterpriseCluster     app.redislabs.com        v1                                               false
redis       RedisEnterpriseDatabase    app.redislabs.com        v1                                               false
weblogic    Domain                     weblogic.oracle          v8         spec.serverStartPolicy,string,NEVER   false

```

File: content/portworx-install-with-kubernetes/storage-operations/stateful-applications/application-registration.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+1. Create an `applicationregistration` spec, specifying the following:
+
+  * **metadata.name:** With the name of the spec. 
+  * **resources.PodsPath:** With The path which stores the pods created by the CR. These will be deleted when scaling down the migration.
+  * **resources.group:** The group of the CRD being registered.
+  * **resources.version:** The version of the CRD being registered.
+  * **resources.kind:** The kind of the CRD being registered. 
+  * **resources.keepStatus:** If you will not be saving the resource's status, set this value to `false`, otherwise...
+  * **resources.suspendOptions.path:** With the path of a spec that runs when Stork stops your resource.
+  * **resources.suspendOptions.type:** With the data type Stork will send to the spec when it stops your resource. <!-- need help on these, not sure if this is right at all -->
```
Comment (disrani-px) [inline content/portworx-install-with-kubernetes/storage-operations/stateful-applications/application-registration.md:57]:
```suggestion
  * **resources.PodsPath (optional):** With The path which stores the pods created by the CR. These will be deleted when scaling down the migration.
  * **resources.group:** The group of the CRD being registered.
  * **resources.version:** The version of the CRD being registered.
  * **resources.kind:** The kind of the CRD being registered. 
  * **resources.keepStatus (optional):** If you don't want to save the resource's status after migration, set this value to `false`
  * **resources.suspendOptions.path (optional):** The path in the CRD spec which contains the option to suspend the application
  * **resources.suspendOptions.type (optional):** The type of the field that is used to suspend the operation (eq int if the field contains the replica count for the application)
```

**Pull request merged**